docker/models/Dockerfile.ctsm

```diff
@@ -0,0 +1,30 @@
+# ----------------------------------------------------------------------
+# BUILD CTSM BINARY
+# ----------------------------------------------------------------------
+FROM centos:centos7 as ctsm-binary
+# temp note: may want to select a specific version number of centos 7 once we have a working version
+# see list of centos versions on docker hub for reference
+
+# Some variables that can be used to set control the docker build
+#ARG MODEL_VERSION= !! ADD SPECIFIC CTSM GIT HASH HERE WHEN APPROPRIATE !!
+
+# install gcc dependencies
+RUN yum update -y \
+    && yum install make cmake \
+    && bzip2 gcc gcc-c++ gcc-fortran \
```

Comment:
I'm assuming you need a version of `gcc` to compile `gcc` itself, but you can probably save a bit of time by skipping `gcc-c++` and `gcc-fortran`, right? Unless you need two different versions for different steps?

Reply:
Good point! I also realized I screwed up the RUN commands with too many &&

Reply:
@ashiklom turns out the compile of gcc 5.5.0 fails if I dont sudo yum install both gcc-c++ or gfortran?

Reply:
I guess you do need it then. I bet it's necessary to compile the language support.